### PR TITLE
add-address-with-google-API

### PR DESCRIPTION
**Pull request by HoangVanTrinh92**

### CODE REVIEW

File: app/controllers/addresses_controller.rb [hunk 1]
```diff
@@ -0,0 +1,30 @@
+class AddressesController < ApplicationController
+
+  def index
+    @addresses = Address.all
+  end
+
+  def show
+    @address = Address.find(params[:id])
```
Comment (DucNguyenVan) [inline app/controllers/addresses_controller.rb]: remove ()

File: app/assets/javascripts/googlemapapi.js [hunk 2]
```diff
@@ -0,0 +1,113 @@
+var map = null;
+var marker = null;
+
+var infowindow = new google.maps.InfoWindow({
+  size: new google.maps.Size(150,50)});
```
Comment (NguyenHai2310) [inline app/assets/javascripts/googlemapapi.js]: use gem_setting

File: app/assets/javascripts/googlemapapi.js [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+
+  var contentString = html;
+
+  var marker = new google.maps.Marker({
+    position: latlng,
+    map: map,
+    zIndex: Math.round(latlng.lat()*-100000)<<5
+    });
+
+  google.maps.event.addListener(marker, 'click', function() {
```
Comment (NguyenHai2310) [inline app/assets/javascripts/googlemapapi.js]: " instead '

File: app/assets/javascripts/googlemapapi.js [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+    map.fitBounds(bounds);
+  });
+  google.maps.event.addListener(map, "bounds_changed", function() {
+    var bounds = map.getBounds();
+    searchBox.setBounds(bounds);
+  });
+
+  google.maps.event.addListener(map, "click", function() {
+    infowindow.close();
+    });
```
Comment (phuong1492) [inline app/assets/javascripts/googlemapapi.js]: ident

File: app/assets/javascripts/googlemapapi.js [hunk 5]
```diff
@@ -0,0 +1,118 @@
...
+
+function createMarker(latlng, name, html) {
+
+  var contentString = html;
+
+  var marker = new google.maps.Marker({
+    position: latlng,
+    map: map,
+    zIndex: Math.round(latlng.lat()*-100000)<<5
+    });
```
Comment (khanhkhanh) [inline app/assets/javascripts/googlemapapi.js]: Indentation (all the file)

File: app/views/addresses/_add_form.html.erb [hunk 6]
```diff
@@ -0,0 +1,24 @@
+<%= form_for (@address) do |f| %>
```
Comment (khanhkhanh) [inline app/views/addresses/_add_form.html.erb]: Remove ()

File: app/views/addresses/new.html.erb [hunk 7]
```diff
@@ -0,0 +1,6 @@
+<div id="left-con">
+  <h3><%= t "address.add_address" %> </h3>
+  <%= render "addresses/add_form" %>
+</div>
+<input id="pac-input" class="controls" type="text" placeholder="Search Box">
```
Comment (khanhkhanh) [inline app/views/addresses/new.html.erb]: i18n

File: app/assets/javascripts/googlemapapi.js [hunk 8]
```diff
@@ -0,0 +1,118 @@
...
+
+  google.maps.event.trigger(marker, "click");
+  return marker;
+}
+
+function initialize() {
+
+  var myLatlng = new google.maps.LatLng(21.017030, 105.783902);
+  var markers = [];
```
Comment (DucNguyenVan) [inline app/assets/javascripts/googlemapapi.js]: follow convention, it should be change to Array.new

File: app/assets/javascripts/googlemapapi.js [hunk 9]
```diff
@@ -0,0 +1,95 @@
...
+      };
+      var marker = new google.maps.Marker({
+        map: map,
+        icon: image,
+        title: place.name,
+        position: place.geometry.location
+      });
+
+      markers.push(marker);
+
```
Comment (daothanhcam) [inline app/assets/javascripts/googlemapapi.js]: Remove this line.

**Pull request merged**